src/main/java/io/confluent/idesidecar/restapi/resources/ConnectionsResource.java

```diff
@@ -83,11 +84,35 @@ public Uni<ConnectionsList> listConnections() {
   @POST
   @Consumes(MediaType.APPLICATION_JSON)
   @Produces(MediaType.APPLICATION_JSON)
-  public Connection createConnection(ConnectionSpec connectionSpec)
-      throws ConnectionNotFoundException, CreateConnectionException {
-
-    var newSpec = connectionStateManager.createConnectionState(connectionSpec);
-    return Connection.from(newSpec);
+  public Uni<Connection> createConnection(
+      ConnectionSpec connectionSpec,
+      @Schema(
+          description = "Whether to check the connection without creating it",
+          defaultValue = "false"
+      )
+      @QueryParam("dry_run")
+      boolean dryRun
+  ) throws ConnectionNotFoundException, CreateConnectionException {
+    if (dryRun) {
+      // Just test the connection and return the status
+      var testedState = connectionStateManager.testConnectionState(connectionSpec);
+      // Get the status of the connection
+      var futureStatus = testedState.getConnectionStatus();
+      // And create a uni that will complete when the status is available
+      return Uni
+          .createFrom()
+          .completionStage(futureStatus.toCompletionStage())
+          .map(connectionStatus ->
+              Connection.from(testedState, connectionStatus)
+          );
+    }
```

Comment:
This is the bulk of the changes in the Connections API to add the `dry_run` query parameter to the `POST /gateway/v1/connections` endpoint.

Reply:
Kudos for making this look straightforward 👍